### PR TITLE
feat(memory): Add RSS reading with experimental sysinfo

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds a new experimental version of `sysinfo` which can read `rss` memory stats of cgroups. This is related to the following PR: https://github.com/GuillaumeGomez/sysinfo/pull/1309

#skip-changelog

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -163,7 +163,7 @@ symbolic-unreal = { version = "12.1.2", default-features = false }
 syn = "1.0.14"
 syn2 = { package = "syn", version = "2.0.11" }
 synstructure = "0.12.3"
-sysinfo = { version = "0.30.7", default-features = false }
+sysinfo = { git = "https://github.com/getsentry/sysinfo.git", rev = "e2e5d530600f96bdd79652c856918da23e5dd938" }
```
Comment (Dav1dde) [inline Cargo.toml:169]: Should add a comment with a link to the PR why we are on the fork (something like I did for redis)

File: relay-server/src/services/health_check.rs [hunk 2]
```diff
@@ -114,19 +114,24 @@ impl HealthCheckService {
...
         let memory = match self.system.cgroup_limits() {
             Some(cgroup) => Memory {
                 used: cgroup.total_memory.saturating_sub(cgroup.free_memory),
                 total: cgroup.total_memory,
+                rss: cgroup.rss,
             },
             None => Memory {
                 used: self.system.used_memory(),
                 total: self.system.total_memory(),
+                rss: self.system.used_memory(),
```
Comment (Dav1dde) [inline relay-server/src/services/health_check.rs:128]: DIdn't you say `total_memory() - available_memory()` is closer to the RSS?

**Review by Dav1dde — APPROVED** [review]

File: relay-server/src/services/health_check.rs (hunk 2 above)
Comment (iambriccardo) [reply relay-server/src/services/health_check.rs:128]: Yeah but used_memory is implemented as such for linux: https://github.com/GuillaumeGomez/sysinfo/blob/3cc3df817d3f3c1118ae8bcc438c5ca1c29c016e/src/unix/linux/system.rs#L380

File: Cargo.toml (hunk 1 above)
Comment (iambriccardo) [reply Cargo.toml:169]: The PR is already linked in the description

Comment (Dav1dde) [reply Cargo.toml:169]: Please add it as a comment to the code, there should be justification in the code not in some document or PR description. If somebody looks at the `Cargo.toml` they should know why we depend on a git dependency without going through an infinite loop of git blames.

**Pull request merged**